### PR TITLE
fix: sell-side keyboard navigation

### PR DESCRIPTION
**Pull request by iprado78**
* Allows tab navigation on credit sell-side ticket, and submitting of quote on key press
* Re-usable hook to programmatically trigger a click on an element when a parent receives an "Enter" key press

### CODE REVIEW

**Review by SHolleworth — DISMISSED** [review]

File: src/client/e2e/credit.spec.ts [hunk 1]
```diff
@@ -92,4 +91,44 @@ test.describe("Credit", () => {
       expect(tradeId).toEqual(blotterId)
     })
   })
+
+  test.describe("Sell side", () => {
+    test.only("Sell side ticket", async ({ context }) => {
```
Comment (algreasley) [inline src/client/e2e/credit.spec.ts:96]: drop the `.only` now?

File: src/client/e2e/credit.spec.ts [hunk 2]
```diff
@@ -4,37 +4,36 @@ import { test } from "./fixtures"
...
+      rfqsPage = newRfqPage
+      rfqBlotterPage = newRfqPage
+    }
+  })
 
-        rfqsPage = newRfqPage
-        rfqBlotterPage = newRfqPage
-      }
+  test.describe("New RFQ", () => {
+    test("When I select Googl instrument and click Send RFQ button then I should see a GOOGL RFQ created on the RFQ sections and I can accept any value @smoke", async () => {
```
Comment (algreasley) [inline src/client/e2e/credit.spec.ts:35]: test label is way too long, isn't it?
"Create RFQ for GOOGL" should be enough ..

File: src/client/e2e/credit.spec.ts [hunk 3]
```diff
@@ -92,4 +91,44 @@ test.describe("Credit", () => {
...
+    test.only("Sell side ticket", async ({ context }) => {
+      await newRfqPage.getByPlaceholder(/Enter a CUSIP/).click()
+      await newRfqPage
+        .locator("[data-testid='search-result-item']")
+        .nth(5)
+        .click()
+
+      const quantity = newRfqPage.locator("[data-testid='quantity']")
+      await quantity.type("2")
+      await quantity.blur()
```
Comment (algreasley) [inline src/client/e2e/credit.spec.ts:105]: minor: wot's the `.blur()` for?

Comment (algreasley) [reply src/client/e2e/credit.spec.ts:105]: side note .. cursor management is ****ed in the Dealer checkbox list ...

File: src/client/e2e/credit.spec.ts [hunk 4]
```diff
@@ -92,4 +91,44 @@ test.describe("Credit", () => {
...
+
+      await newRfqPage
+        .locator("span")
+        .getByText(/Adaptive Bank/)
+        .click()
+
+      const pagePromise = context.waitForEvent("page")
```
Comment (algreasley) [inline src/client/e2e/credit.spec.ts:112]: minor/question .. why is `pagePromise` set up here, and awaited below ..

File: src/client/e2e/credit.spec.ts [hunk 5]
```diff
@@ -92,4 +91,44 @@ test.describe("Credit", () => {
...
+      const sellSidePage = await pagePromise
+
+      await sellSidePage
+        .getByText(/New RFQ/)
+        .first()
+        .click()
+
+      await sellSidePage.getByTestId("price-input").fill("100")
+
+      await sellSidePage.keyboard.press("Tab")
```
Comment (algreasley) [inline src/client/e2e/credit.spec.ts:128]: could even just have Enter here as base test?

File: src/client/src/App/Credit/SellSide/SellSideTradeTicket/SellSideTradeTickerFooter.tsx [hunk 6]
```diff
@@ -145,6 +159,7 @@ export const SellSideTradeTicketFooter = ({
       {state === RfqState.Open && (
         <>
           <PassButton
+            tabIndex={2}
```
Comment (algreasley) [inline src/client/src/App/Credit/SellSide/SellSideTradeTicket/SellSideTradeTickerFooter.tsx:162]: per chat .. reconsider when Pass comes online .. for now, I would remove these `tabIndex` overrides and go with default flow - and disable `PassButton` please (should have been done last year!) .. especially as we have Enter-to-submit-quote in Price field

File: src/client/src/App/Credit/SellSide/SellSideTradeTicket/SellSideTradeTicketParameters.tsx [hunk 7]
```diff
@@ -155,13 +156,19 @@ export const SellSideTradeTicketParameters = ({
           )
         ) : (
           <ParameterInput
+            tabIndex={1}
```
Comment (algreasley) [inline src/client/src/App/Credit/SellSide/SellSideTradeTicket/SellSideTradeTicketParameters.tsx:159]: odd .. what happens if this is not there?

File: src/client/e2e/credit.spec.ts (hunk 1 above)
Comment (SHolleworth) [reply src/client/e2e/credit.spec.ts:96]: oops

File: src/client/e2e/credit.spec.ts (hunk 3 above)
Comment (SHolleworth) [reply src/client/e2e/credit.spec.ts:105]: copy/paste artifact, will remove

File: src/client/e2e/credit.spec.ts (hunk 4 above)
Comment (SHolleworth) [reply src/client/e2e/credit.spec.ts:112]: This is how the docs example looks, I did try initialising it after and it didn't work, I have no idea why it has to be this way

File: src/client/e2e/credit.spec.ts (hunk 5 above)
Comment (SHolleworth) [reply src/client/e2e/credit.spec.ts:128]: yes

File: src/client/e2e/credit.spec.ts (hunk 3 above)
Comment (SHolleworth) [reply src/client/e2e/credit.spec.ts:105]: not sure what you mean by the dealer checkbox list

Comment (algreasley) [reply src/client/e2e/credit.spec.ts]: in the UI .. the list of Dealers that you choose (Adaptive Bank etc.) .. hovering over and moving down the list is flaky (not a now thing, dw)

**Review by algreasley — APPROVED** [review]

**Pull request merged**